### ISSUE TITLE
TrainingParameterScheduleDouble not working as expected? (CNTK C#)

### ISSUE THREAD

**Issue by Valdas3** [Microsoft/CNTK#2812]
I am confused by different overloads of TrainingParameterScheduleDouble. Below is my training loop.

    var trainingLoss = CNTKLib.SquaredError(output, normalizedLabel, "lossFunction");
    var parameterLearners = new List<Learner>() { adagrad };
    var trainer = Trainer.CreateTrainer(output, trainingLoss, null, parameterLearners);
    while (true)
    {
        var minibatchDataHR = miniBatchSourceHR.GetNextMinibatch(1<<10, device);
        var minibatchDataLR = miniBatchSourceLR.GetNextMinibatch(1<<10, device);
        if (minibatchDataHR.empty())
        {
            break;
        }
        trainer.TrainMinibatch(
            new Dictionary<Variable, MinibatchData>{ { inputVariable, minibatchDataLR[imageStreamInfoLR] },
                { labelVariable, minibatchDataHR[imageStreamInfoHR] } },
            device);
        TestHelper.PrintTrainingProgress(trainer, miniBatchCount++, outputFrequencyInMinibatches);
    }
If I create the learner like this then the learning rate remains constant all the time.

    var schedule = new TrainingParameterScheduleDouble(0.0178125, 1);
    var adagrad = CNTKLib.AdaGradLearner(new ParameterVector(output.Parameters().ToList()), schedule);

This is how loss function value changes for first 1000 mini-batches

    Minibatch: 0 SquaredError = 815.367248535156
    Minibatch: 50 SquaredError = 52.749641418457
    Minibatch: 100 SquaredError = 37.6697196960449
    Minibatch: 150 SquaredError = 32.7807846069336
    Minibatch: 200 SquaredError = 31.1735057830811
    Minibatch: 250 SquaredError = 28.383373260498
    Minibatch: 300 SquaredError = 29.6063175201416
    Minibatch: 350 SquaredError = 28.0748424530029
    Minibatch: 400 SquaredError = 24.5268440246582
    Minibatch: 450 SquaredError = 25.1975879669189
    Minibatch: 500 SquaredError = 25.2956199645996
    Minibatch: 550 SquaredError = 23.5030403137207
    Minibatch: 600 SquaredError = 24.3890438079834
    Minibatch: 650 SquaredError = 22.5326042175293
    Minibatch: 700 SquaredError = 24.0747203826904
    Minibatch: 750 SquaredError = 23.6230430603027
    Minibatch: 800 SquaredError = 23.6832656860352
    Minibatch: 850 SquaredError = 21.8520469665527
    Minibatch: 900 SquaredError = 22.5924625396729
    Minibatch: 950 SquaredError = 21.4095993041992
    Minibatch: 1000 SquaredError = 23.6634502410889

If I try to create the learner like this then I expect everything to remain identical for 1000 mini-batches, like in the previous example and then learning rate will decrease to 0.0078125.

    var vector = new VectorPairSizeTDouble
    {
        new PairSizeTDouble(1000, 0.0178125),
        new PairSizeTDouble(1, 0.0078125),
    };
    var schedule = new TrainingParameterScheduleDouble(vector, 1 << 10);

This is how loss function value changes for first 1000 mini-batches

    Minibatch: 0 SquaredError = 815.367248535156
    Minibatch: 50 SquaredError = 729.427856445313
    Minibatch: 100 SquaredError = 654.373107910156
    Minibatch: 150 SquaredError = 526.06201171875
    Minibatch: 200 SquaredError = 402.514953613281
    Minibatch: 250 SquaredError = 296.680236816406
    Minibatch: 300 SquaredError = 236.036636352539
    Minibatch: 350 SquaredError = 172.560089111328
    Minibatch: 400 SquaredError = 134.07893371582
    Minibatch: 450 SquaredError = 113.425094604492
    Minibatch: 500 SquaredError = 105.064552307129
    Minibatch: 550 SquaredError = 97.4012451171875
    Minibatch: 600 SquaredError = 98.1433944702148
    Minibatch: 650 SquaredError = 92.5095062255859
    Minibatch: 700 SquaredError = 95.8705368041992
    Minibatch: 750 SquaredError = 95.7256698608398
    Minibatch: 800 SquaredError = 90.9073791503906
    Minibatch: 850 SquaredError = 91.5558776855469
    Minibatch: 900 SquaredError = 93.2224655151367
    Minibatch: 950 SquaredError = 91.1571044921875
    Minibatch: 1000 SquaredError = 94.1823425292969

As you see the error decreases much slower when using the second approach. I have run these examples multiple times and get similar results every time so I do not think this is a fluke. Why is that happening? Shouldn't I get similar results? Have I misunderstood how the second overload of TrainingParameterScheduleDouble supposed to work?
